Add mcxaxx6 flexcan support

NeilChen93 opened the pull request:
Add flexcan support for frdm_mcxa366,frdm_mcxa266 and frdm_mcxa346

NeilChen93 committed:
drivers: syscon: update syscon driver to support mcxa flexcan clock

Add mcxa flexcan clock support

Signed-off-by: Neil Chen <cheng.chen_1@nxp.com>

henrikbrixandersen reviewed (commented):
Seems like this should have been split into two PRs? One for each HW feature?

henrikbrixandersen reviewed (approved):
Thanks!

PetervdPerk-NXP reviewed (changes requested):
MCXAXX6 supports CAN-FD yet this PR only enables CAN2.0B.
Same issue as #91138
#95097 could be a potential resolution for this issue.

henrikbrixandersen commented:
> MCXAXX6 supports CAN-FD yet this PR only enables CAN2.0B. Same issue as #91138 #95097 could be a potential resolution for this issue.

I don't see this as a reason for blocking the introduction of this PR?

bperseghetti reviewed (changes requested):
I concur with @PetervdPerk-NXP, stating flexcan support means it must support CANFD (and can optionally run/support 2.0B). Not the other way around.

https://docs.nxp.com/bundle/UM12018/page/topics/flexcan_interface.html#

henrikbrixandersen reviewed (changes requested):
Similar to https://github.com/zephyrproject-rtos/zephyr/pull/98900, this PR depends on the functionality proposed in https://github.com/zephyrproject-rtos/zephyr/pull/99483 being merged first.

henrikbrixandersen reviewed (changes requested):
Please rebase on `main` and resolve any merge conflicts.

NeilChen93 committed:
tests: drivers: can: timing: enable full timing test

Enable the full range of CAN timing tests on the NXP FRDM-MCXA366,
FRDM-MCXA346 and FRDM-MCXA266 boards.

Signed-off-by: Neil Chen <cheng.chen_1@nxp.com>

str4t0m reviewed (dismissed):
I guess the condition on the specific soc CONFIG_SOC_MCXA346 in the syscon driver is unavoidable, as the HAL mixes the paramter(-less) versions? Needs a rebase otherwise should be fine.

NeilChen93 committed:
dts: arm/nxp: Add Flexcan nodes to NXP mcxaxx6 dtsi file

Add Flexcan0,Flexcan1 node to NXP MCXA366,MCXA266 dtsi file
Add Flexcan0 node to NXP MCXA346 dtsi file

Signed-off-by: Neil Chen <cheng.chen_1@nxp.com>

NeilChen93 committed:
boards: nxp/frdm_mcxaxx6: Support Flexcan for NXP frdm_mcxaxx6 board

Support Flexcan for NXP frdm_mcxa366,frdm_mcxa346,frdm_mcxa266 board

Signed-off-by: Neil Chen <cheng.chen_1@nxp.com>

lucien-nxp commented on dts/arm/nxp/nxp_mcxaxx6_common.dtsi:
```diff
@@ -554,6 +554,30 @@
+			interrupts = <19 0>;
+			interrupt-names = "common";
+			clocks = <&syscon MCUX_FLEXCAN0_CLK>;
+			clk-source = <0>;
```
[henrikbrixandersen](https://github.com/henrikbrixandersen) help improve the flexcan clock settings #102069 . After his PR merged, I think this` clk-source ` property can be removed in soc.dtsi.

lucien-nxp reviewed (commented):
[henrikbrixandersen](https://github.com/henrikbrixandersen) help improve the flexcan clock setting on #102069.

lucien-nxp reviewed (commented):
[henrikbrixandersen](https://github.com/henrikbrixandersen) help improve the flexcan clock settings https://github.com/zephyrproject-rtos/zephyr/pull/102069 . After his PR merged, I think this `clk-source` property can be removed in your patch.

henrikbrixandersen commented on dts/arm/nxp/nxp_mcxaxx6_common.dtsi:
```diff
@@ -554,6 +554,30 @@
+			interrupts = <19 0>;
+			interrupt-names = "common";
+			clocks = <&syscon MCUX_FLEXCAN0_CLK>;
+			clk-source = <0>;
```
Now that https://github.com/zephyrproject-rtos/zephyr/pull/102069 was merged, please remove the `clk-source` property from here.

NeilChen93 replied:
Updated.

henrikbrixandersen reviewed (approved):
Thanks!

str4t0m reviewed (approved):
The remaining blocking reviews have requested to enable CAN-FD support, which has been added since a while.
